great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -169,55 +163,40 @@ def remove_key(self, key: Tuple[str, ...]) -> None:
         self._save_changes()
 
     def _has_key(self, key: Tuple[str, ...]) -> bool:
-        (
-            resource_type,
-            resource_name,
-        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
+        resource_name = InlineStoreBackend._determine_resource_name(key)
+        resource_type = self._resource_type
 
-        if len(key) == 1:
+        if len(key) == 0:
             return resource_type in self._data_context.config
-        elif len(key) == 2:
+        elif len(key) == 1:
             res: dict = self._data_context.config.get(resource_type) or {}
             return resource_name in res
 
         return False
 
-    def _validate_key(self, key: Tuple[str, ...]) -> None:
-        (
-            resource_type,
-            _,
-        ) = InlineStoreBackend._determine_resource_type_and_name_from_key(key)
-
-        if not isinstance(
-            resource_type, DataContextVariableSchema
-        ) and not DataContextVariableSchema.has_value(resource_type):
-            raise TypeError(
-                f"Keys in {self.__class__.__name__} must adhere to the schema defined by {DataContextVariableSchema.__name__}; invalid value ({resource_type}) of type {type(resource_type)} found"
-            )
+    # def _validate_key(self, key: Tuple[str, ...]) -> None:
+    #     if not isinstance(
+    #         resource_type, DataContextVariableSchema
+    #     ) and not DataContextVariableSchema.has_value(resource_type):
+    #         raise TypeError(
+    #             f"Keys in {self.__class__.__name__} must adhere to the schema defined by {DataContextVariableSchema.__name__}; invalid value ({resource_type}) of type {type(resource_type)} found"
+    #         )
 
     def _save_changes(self) -> None:
         # NOTE: <DataContextRefactor> This responsibility will be moved into DataContext Variables object
         self._data_context._save_project_config()
 
     @staticmethod
-    def _determine_resource_type_and_name_from_key(
-        key: Tuple[str, ...]
-    ) -> Tuple[DataContextVariableSchema, Optional[str]]:
-        resource_type: DataContextVariableSchema = DataContextVariableSchema(key[0])
-        resource_name: Optional[str] = None
-        if len(key) > 1 and key[1]:
-            resource_name = key[1]
-
-        return resource_type, resource_name
+    def _determine_resource_name(key: Tuple[str, ...]) -> Optional[str]:
+        resource_name: Optional[str] = key[0] or None
+        return resource_name
```

Comment:
Refactor to leverage rich type (key) would make this a lot cleaner than tuple manipulation - possibly add explicit fork in `store.py` akin to how we're thinking about `GeCloudStoreBackend`

Reply:
Will have to come back to this in a subsequent PR - seems like a decent bit was breaking and this gets us across the line in terms of desired behavior.